1-JavaScript/Week2/homeworkSolition/ex7-sol-mindPrivacy.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+const employeeRecords = [
+  {
+    name: 'John',
+    occupation: 'developer',
+    gender: 'M',
+    email: 'john.doe@somewhere.net',
+    salary: 50000,
+  },
+  {
+    name: 'Jane',
+    occupation: 'manager',
+    gender: 'F',
+    email: 'jane.eyre@somewhere.net',
+    salary: 60000,
+  },
+];
+
+
+function filterPrivateData(empRecords) {
+
+  const generalInfo =[]
+  for(let records of empRecords){
+    // console.log(records)
+    // destructering and pushing each of proporty
+    const {name,occupation,gender,salary,email}=records
+    generalInfo.push({name,occupation,email})
+  }
+  // console.log("Result is",generalInfo)
+  return generalInfo
+
+}
```

Comment:
@robzi18 Nice job with destructuring and using object property shorthand! 🤩 

However, you don't need to grab the properties that you are not going to use: gender and salary.

```suggestion
function filterPrivateData(empRecords) {

  const generalInfo =[]
  for(let records of empRecords){
    // destructering and pushing each of proporty
    const {name, occupation, email}=records
    generalInfo.push({name, occupation, email})
  }
  return generalInfo
}
```

Try and use the `.map()` method for this assignment, then you also don't need to make a new variable `generalInfo`. This array method is especially made this scenarios like this where you want to return a new array and make some modifications. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map